### PR TITLE
Async run through calibration with threads via TaskRunner module from @patmarion

### PR DESCRIPTION
**Pull request by peteflorence**
This PR brings in the TaskRunner class (written by @patmarion -- thanks Pat!) in as a Spartan module.  Can later see if can get this module to live in Director.

It is very useful for wrapping a simple, very readable run of blocking code (written by @manuelli) that operates the calibration step by step.

Calibration now suggested to be started via `cal.runThreaded()`

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/spartan/164)

### CODE REVIEW

File: modules/spartan/taskrunner.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+from director.timercallback import TimerCallback
+
+
+class TaskRunner(object):
+
+  def __init__(self):
+    self.interval = 1/60.0
+    sys.setcheckinterval(1000)
+    #sys.setswitchinterval(self.interval)			# sys.setswitchinterval is only Python 3
+    self.task_queue = asynctaskqueue.AsyncTaskQueue()
```
Comment (patmarion) [inline modules/spartan/taskrunner.py]: btw, if someone is motivated to add this file to director, that would be a welcome PR.  But I would want to fix the style to use 4 space indent and camelCase for variable names.

**Comment by manuelli** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[modules/spartan/taskrunner.py, line 16 at r1](https://reviewable.io:443/reviews/robotlocomotion/spartan/164#-KwHJ-WJcCtlAncc95to-r1-16:-KwHdlF9gqm6GjN3wY6j:bvwtrhl) ([raw file](https://github.com/robotlocomotion/spartan/blob/6d929b243b004c940fc864210f15b20d7163012b/modules/spartan/taskrunner.py#L16)):*
<details><summary><i>Previously, patmarion (Pat Marion) wrote…</i></summary><blockquote>

btw, if someone is motivated to add this file to director, that would be a welcome PR.  But I would want to fix the style to use 4 space indent and camelCase for variable names.
</blockquote></details>

I can work on getting this into director this weekend. It's a super useful bit of functionality.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/spartan/164)*

**Comment by manuelli** [comment]
:lgtm:

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/spartan/164#-:-KwHfDvF11-frJYwhOT9:bnfp4nl)*

**Comment by manuelli** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[modules/spartan/taskrunner.py, line 16 at r1](https://reviewable.io:443/reviews/robotlocomotion/spartan/164#-KwHJ-WJcCtlAncc95to-r1-16:-KwHfLfCxuySDyjEf9lG:b32w0x) ([raw file](https://github.com/robotlocomotion/spartan/blob/6d929b243b004c940fc864210f15b20d7163012b/modules/spartan/taskrunner.py#L16)):*
<details><summary><i>Previously, manuelli (Lucas Manuelli) wrote…</i></summary><blockquote>

I can work on getting this into director this weekend. It's a super useful bit of functionality.
</blockquote></details>

OK

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/spartan/164)*

**Comment by manuelli** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/spartan/164)*

**Pull request merged**